### PR TITLE
reword/cleanup a bit the "why" documentation page

### PR DESCRIPTION
**Pull request by aabadie**
Another documentation cleanup/rewording PR:
* replace other occurences of `we` by more neutral expressions
* remove extra title underline characters (=, -, .)

### CODE REVIEW

File: doc/why.rst [hunk 1]
```diff
@@ -1,16 +1,16 @@
 
 Why joblib: project goals
-===========================
+=========================
 
 What pipelines bring us
---------------------------
```
Comment (GaelVaroquaux) [inline doc/why.rst]: I would completely change this title to "Benefits of pipelines"

File: doc/why.rst [hunk 2]
```diff
@@ -19,33 +19,32 @@ Data-flow programming for performance
   purely functional programming).
 
 Provenance tracking for understanding the code
-...............................................
```
Comment (GaelVaroquaux) [inline doc/why.rst]: This title should be changed in "Provenance tracking to understand the code". My level of English got better in 10 years.

File: doc/why.rst [hunk 3]
```diff
@@ -19,33 +19,32 @@ Data-flow programming for performance
...
+..............................................
 
-* **Tracking of data and computations:** to be able to fully reproduce a
-  computational experiment: requires tracking of the data and operation
-  implemented.
+* **Tracking of data and computations:** This allows the reproducibility of a
```
Comment (GaelVaroquaux) [inline doc/why.rst]: "allows" => "enables".

**Comment by aabadie** [comment]
comments addressed

**Pull request merged**